In [4]:
import numpy as np;
import pandas as pd;
from pandas import DataFrame

In [5]:
data=pd.read_csv("/content/zomato.csv",encoding="latin1")

In [6]:
data.head()

,Restaurant ID,Restaurant Name,Country Code,City,Address,Locality,Locality Verbose,Longitude,Latitude,Cuisines,...,Currency,Has Table booking,Has Online delivery,Is delivering now,Switch to order menu,Price range,Aggregate rating,Rating color,Rating text,Votes
0,6317637,Le Petit Souffle,162,Makati City,"Third Floor, Century City Mall, Kalayaan Avenu...","Century City Mall, Poblacion, Makati City","Century City Mall, Poblacion, Makati City, Mak...",121.027535,14.565443,"French, Japanese, Desserts",...,Botswana Pula(P),Yes,No,No,No,3,4.8,Dark Green,Excellent,314
1,6304287,Izakaya Kikufuji,162,Makati City,"Little Tokyo, 2277 Chino Roces Avenue, Legaspi...","Little Tokyo, Legaspi Village, Makati City","Little Tokyo, Legaspi Village, Makati City, Ma...",121.014101,14.553708,Japanese,...,Botswana Pula(P),Yes,No,No,No,3,4.5,Dark Green,Excellent,591
2,6300002,Heat - Edsa Shangri-La,162,Mandaluyong City,"Edsa Shangri-La, 1 Garden Way, Ortigas, Mandal...","Edsa Shangri-La, Ortigas, Mandaluyong City","Edsa Shangri-La, Ortigas, Mandaluyong City, Ma...",121.056831,14.581404,"Seafood, Asian, Filipino, Indian",...,Botswana Pula(P),Yes,No,No,No,4,4.4,Green,Very Good,270
3,6318506,Ooma,162,Mandaluyong City,"Third Floor, Mega Fashion Hall, SM Megamall, O...","SM Megamall, Ortigas, Mandaluyong City","SM Megamall, Ortigas, Mandaluyong City, Mandal...",121.056475,14.585318,"Japanese, Sushi",...,Botswana Pula(P),No,No,No,No,4,4.9,Dark Green,Excellent,365
4,6314302,Sambo Kojin,162,Mandaluyong City,"Third Floor, Mega Atrium, SM Megamall, Ortigas...","SM Megamall, Ortigas, Mandaluyong City","SM Megamall, Ortigas, Mandaluyong City, Mandal...",121.057508,14.584450,"Japanese, Korean",...,Botswana Pula(P),Yes,No,No,No,4,4.8,Dark Green,Excellent,229


In [7]:
data.shape

(9551, 21)

In [8]:
data.columns

Index(['Restaurant ID', 'Restaurant Name', 'Country Code', 'City', 'Address',
       'Locality', 'Locality Verbose', 'Longitude', 'Latitude', 'Cuisines',
       'Average Cost for two', 'Currency', 'Has Table booking',
       'Has Online delivery', 'Is delivering now', 'Switch to order menu',
       'Price range', 'Aggregate rating', 'Rating color', 'Rating text',
       'Votes'],
      dtype='object')

In [9]:
# Deleting unnecessary columns from the data

cols_to_del = [ 'Country Code', 'Address',
       'Locality', 'Locality Verbose', 'Longitude', 'Latitude' ,'Switch to order menu', 'Rating color' ]

data = data.drop(cols_to_del, axis = 1 )
data.head()


,Restaurant ID,Restaurant Name,City,Cuisines,Average Cost for two,Currency,Has Table booking,Has Online delivery,Is delivering now,Price range,Aggregate rating,Rating text,Votes
0,6317637,Le Petit Souffle,Makati City,"French, Japanese, Desserts",1100,Botswana Pula(P),Yes,No,No,3,4.8,Excellent,314
1,6304287,Izakaya Kikufuji,Makati City,Japanese,1200,Botswana Pula(P),Yes,No,No,3,4.5,Excellent,591
2,6300002,Heat - Edsa Shangri-La,Mandaluyong City,"Seafood, Asian, Filipino, Indian",4000,Botswana Pula(P),Yes,No,No,4,4.4,Very Good,270
3,6318506,Ooma,Mandaluyong City,"Japanese, Sushi",1500,Botswana Pula(P),No,No,No,4,4.9,Excellent,365
4,6314302,Sambo Kojin,Mandaluyong City,"Japanese, Korean",1500,Botswana Pula(P),Yes,No,No,4,4.8,Excellent,229


In [10]:
data.shape

(9551, 13)

In [11]:
data["Rating text"].value_counts()

,count
Rating text,
Average,3737
Not rated,2148
Good,2100
Very Good,1079
Excellent,301
Poor,186


In [12]:
data["Rating text"].value_counts(normalize=True)*100

,proportion
Rating text,
Average,39.126793
Not rated,22.489792
Good,21.987226
Very Good,11.297246
Excellent,3.151502
Poor,1.947440


**Observations**

Lot of Restaurants have been rated as avg (39.12%) only 3.15% of restaurants as Excellent

In [13]:
data[data["Aggregate rating"]>=4]

,Restaurant ID,Restaurant Name,City,Cuisines,Average Cost for two,Currency,Has Table booking,Has Online delivery,Is delivering now,Price range,Aggregate rating,Rating text,Votes
0,6317637,Le Petit Souffle,Makati City,"French, Japanese, Desserts",1100,Botswana Pula(P),Yes,No,No,3,4.8,Excellent,314
1,6304287,Izakaya Kikufuji,Makati City,Japanese,1200,Botswana Pula(P),Yes,No,No,3,4.5,Excellent,591
2,6300002,Heat - Edsa Shangri-La,Mandaluyong City,"Seafood, Asian, Filipino, Indian",4000,Botswana Pula(P),Yes,No,No,4,4.4,Very Good,270
3,6318506,Ooma,Mandaluyong City,"Japanese, Sushi",1500,Botswana Pula(P),No,No,No,4,4.9,Excellent,365
4,6314302,Sambo Kojin,Mandaluyong City,"Japanese, Korean",1500,Botswana Pula(P),Yes,No,No,4,4.8,Excellent,229
...,...,...,...,...,...,...,...,...,...,...,...,...,...
9545,5915054,Baltazar,ÛÁstanbul,"Burger, Izgara",90,Turkish Lira(TL),No,No,No,3,4.3,Very Good,870
9546,5915730,NamlÛ± Gurme,ÛÁstanbul,Turkish,80,Turkish Lira(TL),No,No,No,3,4.1,Very Good,788
9547,5908749,Ceviz AÛôacÛ±,ÛÁstanbul,"World Cuisine, Patisserie, Cafe",105,Turkish Lira(TL),No,No,No,3,4.2,Very Good,1034
9549,5916112,Aôôk Kahve,ÛÁstanbul,Restaurant Cafe,120,Turkish Lira(TL),No,No,No,4,4.0,Very Good,901


In [14]:
data[["Restaurant Name","Cuisines"]]

,Restaurant Name,Cuisines
0,Le Petit Souffle,"French, Japanese, Desserts"
1,Izakaya Kikufuji,Japanese
2,Heat - Edsa Shangri-La,"Seafood, Asian, Filipino, Indian"
3,Ooma,"Japanese, Sushi"
4,Sambo Kojin,"Japanese, Korean"
...,...,...
9546,NamlÛ± Gurme,Turkish
9547,Ceviz AÛôacÛ±,"World Cuisine, Patisserie, Cafe"
9548,Huqqa,"Italian, World Cuisine"
9549,Aôôk Kahve,Restaurant Cafe


In [15]:
data["Currency"].unique()

array(['Botswana Pula(P)', 'Brazilian Real(R$)', 'Dollar($)',
       'Emirati Diram(AED)', 'Indian Rupees(Rs.)',
       'Indonesian Rupiah(IDR)', 'NewZealand($)', 'Pounds(\x8c£)',
       'Qatari Rial(QR)', 'Rand(R)', 'Sri Lankan Rupee(LKR)',
       'Turkish Lira(TL)'], dtype=object)

In [16]:
data[data["Currency"]=="Indian Rupees(Rs.)"]

,Restaurant ID,Restaurant Name,City,Cuisines,Average Cost for two,Currency,Has Table booking,Has Online delivery,Is delivering now,Price range,Aggregate rating,Rating text,Votes
624,3400025,Jahanpanah,Agra,"North Indian, Mughlai",850,Indian Rupees(Rs.),No,No,No,3,3.9,Good,140
625,3400341,Rangrezz Restaurant,Agra,"North Indian, Mughlai",700,Indian Rupees(Rs.),No,No,No,2,3.5,Good,71
626,3400005,Time2Eat - Mama Chicken,Agra,North Indian,500,Indian Rupees(Rs.),No,No,No,2,3.6,Good,94
627,3400021,Chokho Jeeman Marwari Jain Bhojanalya,Agra,Rajasthani,400,Indian Rupees(Rs.),No,No,No,2,4.0,Very Good,87
628,3400017,Pinch Of Spice,Agra,"North Indian, Chinese, Mughlai",1000,Indian Rupees(Rs.),No,No,No,3,4.2,Very Good,177
...,...,...,...,...,...,...,...,...,...,...,...,...,...
9271,2800100,D Cabana,Vizag,"Continental, Seafood, Chinese, North Indian, B...",600,Indian Rupees(Rs.),No,No,No,2,3.6,Good,193
9272,2800418,Kaloreez,Vizag,"Cafe, North Indian, Chinese",400,Indian Rupees(Rs.),No,No,No,2,3.7,Good,85
9273,2800881,Plot 17,Vizag,"Burger, Pizza, Biryani",600,Indian Rupees(Rs.),No,No,No,2,4.3,Very Good,172
9274,2800042,Vista - The Park,Vizag,"American, North Indian, Thai, Continental",1500,Indian Rupees(Rs.),No,No,No,4,3.8,Good,74


In [19]:
data[(data["Currency"] == "Indian Rupees(Rs.)") & (data["City"] == "Vizag")]

,Restaurant ID,Restaurant Name,City,Cuisines,Average Cost for two,Currency,Has Table booking,Has Online delivery,Is delivering now,Price range,Aggregate rating,Rating text,Votes
9256,2800013,Pizza Hut,Vizag,"Pizza, Fast Food",600,Indian Rupees(Rs.),No,No,No,2,4.6,Excellent,289
9257,2800012,Pizza Hut,Vizag,"Pizza, Fast Food",600,Indian Rupees(Rs.),No,No,No,2,4.3,Very Good,230
9258,2800294,Sri Sairam Parlour,Vizag,"South Indian, North Indian",300,Indian Rupees(Rs.),No,No,No,1,4.2,Very Good,270
9259,2800903,Mekong - Hotel GreenPark,Vizag,"Chinese, Thai, Burmese, Vietnamese, Tibetan, J...",1300,Indian Rupees(Rs.),No,No,No,3,4.4,Very Good,73
9260,2800052,The Square - Hotel Novotel,Vizag,"Continental, North Indian",1700,Indian Rupees(Rs.),No,No,No,4,4.1,Very Good,125
9261,2800095,Alpha Hotel,Vizag,"North Indian, Biryani",300,Indian Rupees(Rs.),No,No,No,1,3.7,Good,240
9262,2800856,Barbeque Nation,Vizag,"North Indian, Chinese, Mediterranean",1600,Indian Rupees(Rs.),No,No,No,4,4.9,Excellent,345
9263,2800911,Double Roti,Vizag,"Cafe, Fast Food, American",1000,Indian Rupees(Rs.),No,No,No,3,3.8,Good,27
9264,2800757,Bake My Wish,Vizag,Fast Food,450,Indian Rupees(Rs.),No,No,No,2,3.8,Good,175
9265,2800096,Kamat Restaurant,Vizag,"Andhra, North Indian, Chinese, Seafood, Biryani",600,Indian Rupees(Rs.),No,No,No,2,3.6,Good,154
